fails:Enumerator::Chain#initialize is a private method
fails:Enumerator::Chain#initialize returns self
fails:Enumerator::Chain#initialize accepts many arguments
fails:Enumerator::Chain#initialize on frozen instance raises a RuntimeError
